src/mad_spark_alt/evolution/genetic_algorithm.py

```
@@ -465,26 +516,97 @@
                 # If no crossover, just copy parents
                 offspring1, offspring2 = parents[0].idea, parents[1].idea
 
-            # Mutation
-            offspring1 = await self.mutation_operator.mutate(
-                offspring1, config.mutation_rate, context
+            # Mutation for offspring1 (with smart semantic selection)
+            # Always pass to mutation operator - it handles probability internally
+            # Create a temporary IndividualFitness for smart selection
+            temp_individual = IndividualFitness(
+                idea=offspring1,
+                overall_fitness=0.8  # High fitness to pass smart selector threshold
```

Ruff (0.12.2)

Trailing comma missing

Add trailing comma

(COM812)